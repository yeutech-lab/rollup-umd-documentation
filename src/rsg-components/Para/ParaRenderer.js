import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

export const styles = ({ space, color, fontFamily }) => ({
	para: {
		marginTop: 0,
		marginBottom: space[2],
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: 'inherit',
		lineHeight: 1.5,
	},
});

export function ParaRenderer({ classes, semantic, children }) {
	const Tag = semantic || 'div';

	return <Tag className={classes.para}>{children}</Tag>;
}

ParaRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	semantic: PropTypes.oneOf(['p']),
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(ParaRenderer);
