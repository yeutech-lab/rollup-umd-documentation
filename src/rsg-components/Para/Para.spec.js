import React from 'react';
import { ParaRenderer, styles } from './ParaRenderer';

const props = {
	classes: classes(styles),
};

it('should render paragraph as a <div>', () => {
	const actual = shallow(<ParaRenderer {...props}>Pizza</ParaRenderer>);

	expect(actual).toMatchSnapshot();
});

it('should render paragraph as a <p>', () => {
	const actual = shallow(
		<ParaRenderer {...props} semantic="p">
			Pizza
		</ParaRenderer>
	);

	expect(actual).toMatchSnapshot();
});
