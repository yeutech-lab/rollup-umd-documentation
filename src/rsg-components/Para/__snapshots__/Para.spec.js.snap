// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render paragraph as a <div> 1`] = `
<div
  className="para"
>
  Pizza
</div>
`;

exports[`should render paragraph as a <p> 1`] = `
<p
  className="para"
>
  Pizza
</p>
`;
